front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/AdminController.java

```diff
@@ -42,5 +43,15 @@ public AdminController(Collection<AdminOperations> adminOperations) {
   @RequestMapping(value = "/recover", method = RequestMethod.POST)
   void recover(@RequestBody AdminOperations.Recover operation) {
     adminOperations.forEach(o -> o.recover(operation));
+    // If we are recovering application , recover application permission also. Without that
```

Comment:
Suggested wording change (Also avoids the ` , ` which looks a little strange in the comment).

```
// Application permissions need to be recovered alongside the application itself
```

In the future, it can be a little easier to read if we line break on full sentences.

```
// If we are recovering application, recover application permission also.
// Without that subsequent CRUD operations on application will fail.
```